### ISSUE TITLE
ONNX --> numpy or txt file

### ISSUE THREAD

**Issue by elboyran** [NLeSC/XAI#33]
Figure out how to convert ONNX to numpy or txt file

**Comment by elboyran** [comment]
See the notebook in onnx examples called `print_onnx.ipynb`

**Comment by elboyran** [comment]
Using onnxmltools produces the same (visually) JSON format as the pretty printing. But it is a proper JSON file.

**Comment by elboyran** [comment]
The onnxmltools were installed in `onnx1.3_onnxmltools_python3.6 `conda environment. This started as a copy of the `onnx1.3_python3.6` environment. onnxmltools is simply installed via pypi.

**Comment by alrevuelta** [comment]
Just FYI, looks like `onnxmltools` is deprecating `save_text`. See merged pull request [https://github.com/onnx/onnxmltools/pull/277](link)

**Comment by ssabatier** [comment]
Hello. Were you ever able to be able to read an .onnx file into the LRP toolbox? I tried converting to a .json with `save_text` and directly converting the onnx_model to pickle with `onnx.load('model.onnx')` but it seems the only way to use custom model files is to have to convert to .txt to be able to convert the model to LRP toolbox's `Sequential` object.

**Comment by elboyran** [comment]
> 
> 
> Hello. Were you ever able to be able to read an .onnx file into the LRP toolbox? I tried converting to a .json with `save_text` and directly converting the onnx_model to pickle with `onnx.load('model.onnx')` but it seems the only way to use custom model files is to have to convert to .txt to be able to convert the model to LRP toolbox's `Sequential` object.

Hi, I haven't worked on this for few years now, might pick it up soon-ish. No, I haven't tried what you are asking, indeed I don't think it works. BTW they have made (already a bit old) a new bigger toolbox: [iNNvestigate](https://github.com/albermax/innvestigate)

**Comment by ssabatier** [comment]
Thank you for the speedy reply and for the info on iNNvestigate. I've dived a little deeper into it and it seems the model has to be converted to Keras to work properly with iNNvestigate. Thus arises a new problem haha. Using https://pypi.org/project/onnx2keras/ seems straightforward, but since the original model was created in PyTorch I'm hitting some snags in recreation specifically with: `Negative dimension size caused by subtracting 7 from 6 for '{{node 341/Conv2D_8}} = Conv2D[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], explicit_paddings=[], padding="VALID", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true](341_pad/Pad_8, 341/Conv2D_8/ReadVariableOp)' with input shapes: [?,3,6,6], [7,7,3,16].` Unfortunately there's no way for me to change the model. I think there is an issue with the input but since the onnx2keras package doesn't require input to convert I'm a little dumbfounded. If I make any progress I will let you know. I look forward to any insights you can provide when you pick the project back up. Thank you again.
